neptune/new/attributes/atoms/string.py

```diff
@@ -13,19 +13,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-from typing import Union
+from typing import List, TYPE_CHECKING, Union
 
+import click
 
 from neptune.new.internal.operation import AssignString
+from neptune.new.internal.utils.paths import path_to_str
 from neptune.new.types.atoms.string import String as StringVal
 from neptune.new.attributes.atoms.atom import Atom
 
+if TYPE_CHECKING:
+    from neptune.new.attribute_container import AttributeContainer
+
 
 class String(Atom):
+
+    MAX_VALUE_LENGTH = 16384
+
+    def __init__(self, container: "AttributeContainer", path: List[str]):
+        super().__init__(container, path)
+        self._value_truncation_occurred = False
+
     def assign(self, value: Union[StringVal, str], wait: bool = False):
         if not isinstance(value, StringVal):
             value = StringVal(value)
 
+        if (
+            not self._value_truncation_occurred
+            and len(value.value) > String.MAX_VALUE_LENGTH
+        ):
+            # the first truncation
+            self._value_truncation_occurred = True
+            click.echo(
+                f"Warning: string '{path_to_str(self._path)}' value was "
+                f"longer than {String.MAX_VALUE_LENGTH} characters and was truncated. "
+                f"This warning is printed only once.",
+                err=True,
+            )
+        value.value = value.value[: String.MAX_VALUE_LENGTH]
```

Comment:
It looks like NOP when previous' if expression is `False` but still I'd prefer truncation to be inside `if` block.